lib/Common/Memory/Recycler.cpp

```diff
@@ -3342,54 +3342,60 @@ Recycler::CollectWithHeuristic()
 {
     // CollectHeuristic_Never flag should only be used with exhaustive candidate
     Assert((flags & CollectHeuristic_Never) == 0);
-
+    
+    const BOOL isScriptContextCloseGCPending = this->collectionWrapper->GetAndClearIsScriptContextCloseGCPending();
     const BOOL allocSize = flags & CollectHeuristic_AllocSize;
     const BOOL timedIfScriptActive = flags & CollectHeuristic_TimeIfScriptActive;
     const BOOL timedIfInScript = flags & CollectHeuristic_TimeIfInScript;
     const BOOL timed = (timedIfScriptActive && isScriptActive) || (timedIfInScript && isInScript) || (flags & CollectHeuristic_Time);
 
-#if ENABLE_PARTIAL_GC
-    if (GetPartialFlag<flags>())
+    // If there is a script context close GC pending, we need to do a GC regardless
+    // Otherwise, we should check the heuristics to see if a GC is necessary
+    if (!isScriptContextCloseGCPending)
```

Comment:
>isScriptContextCloseGCPending [](start = 9, length = 29)

CollectWithHeuristic function gets indirectly called from CollectNow at many places and by default the value of isScriptContextCloseGCPending is true does that mean this if will be skipped most of the time?